test/unittest/tests/block/test_block_session.cpp

```diff
@@ -0,0 +1,270 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+/*
+ * [block_session]: block_session.c
+ * The block manager extent list consists of both extent and size type blocks. This unit test
+ * suite tests aims to test all of the allocation and frees of the extent and size block functions.
+ *
+ * The block session manages an internal caching mechanism for both block and size blocks that are
+ * created or discarded.
+ */
+#include "wt_internal.h"
+#include <catch2/catch.hpp>
+#include "../wrappers/mock_session.h"
+
+void
+cleanup_ext_list(WT_BLOCK_MGR_SESSION *bms)
+{
+    WT_EXT *curr = bms->ext_cache;
+    while (curr != nullptr) {
+        WT_EXT *tmp = curr;
+        curr = curr->next[0];
+        __wt_free(nullptr, tmp);
+    }
+}
+
+void
+validate_ext_list(WT_BLOCK_MGR_SESSION *bms, int expected_items)
+{
+    REQUIRE(bms != nullptr);
+    WT_EXT *curr = bms->ext_cache;
+
+    if (bms->ext_cache_cnt == 0)
+        REQUIRE(bms->ext_cache == nullptr);
+
+    REQUIRE(bms->ext_cache_cnt == expected_items);
+    for (int i = 0; i < expected_items; i++) {
+        REQUIRE(bms->ext_cache != nullptr);
```

Comment:
For each WT_EXT you walk you should call `validate_ext_block` on it. If you split functions out a bit more you can do this with 6 functions:

1. validate_ext_block
2. free_ext_block
3. validate_and_free_ext_block
4. validate_ext_list
5. free_ext_list
6. validate_and_free_ext_list

This would look like this:
```
void
free_ext_block(WT_EXT *ext) {
    __wt_free(nullptr, ext);
}

void
validate_ext_block(WT_EXT *ext)
{
    REQUIRE(ext != nullptr);
    REQUIRE(ext->depth != 0);
    REQUIRE(ext->size == 0);
    REQUIRE(ext->off == 0);

    for (int i = 0; i < ext->depth; i++) {
        REQUIRE(ext->next[i + ext->depth] == nullptr);
    }
}

void
validate_and_free_ext_block(WT_EXT *ext) {
    validate_ext_block(ext);
    free_ext_block(ext);
}

void
free_ext_list(WT_BLOCK_MGR_SESSION *bms)
{
    WT_EXT *curr = bms->ext_cache;
    while (curr != nullptr) {
        WT_EXT *tmp = curr;
        curr = curr->next[0];
        free_ext_block(tmp);
    }
}

void
validate_ext_list(WT_BLOCK_MGR_SESSION *bms, int expected_items)
{
    REQUIRE(bms != nullptr);
    WT_EXT *curr = bms->ext_cache;

    if (bms->ext_cache_cnt == 0)
        REQUIRE(bms->ext_cache == nullptr);

    REQUIRE(bms->ext_cache_cnt == expected_items);
    for (int i = 0; i < expected_items; i++) {
        validate_ext_block(curr);
        REQUIRE(bms->ext_cache != nullptr);
        curr = curr->next[0];
    }
    REQUIRE(curr == nullptr);
}
```

Reply:
Makes sense, i have done that now thanks!